xlators/features/shard/src/shard.c

```diff
@@ -1064,23 +1067,23 @@ shard_common_resolve_shards(call_frame_t *frame, xlator_t *this,
     else
         gf_uuid_copy(gfid, local->base_gfid);
 
+    /* Build base shard path before appending index of the shard */
+    SHARD_BUILD_BASE_PATH(path, gfid, prefix_len);
+
     while (shard_idx_iter <= resolve_count) {
         i++;
         if (shard_idx_iter == 0) {
             local->inode_list[i] = inode_ref(res_inode);
             shard_idx_iter++;
             continue;
         }
-
-        shard_make_block_abspath(shard_idx_iter, gfid, path, sizeof(path));
```

Comment:
Do we need shard_make_block_abspath() in the code base any more?

Reply:
Not needed as of now. Thought to remove it but just that if somebody wants to use it. Anyway, I will remove it in the next patch update